rfcs/detecting_sensitive_textual_content/20230309-project_proposal_detecting_sensitive_textual_content.md

```diff
@@ -0,0 +1,379 @@
+search with the goal of unifying the approaches in the future.
+
+### Frontend
+
+Sensitive results never appear for users who have not opted-in to including
+sensitive results in their query. This feature will be built off the existing
+"mature" filter but enhanced with better UI and more comprehensive/less
+suggestive language. The API query parameter is not present in the frontend
+search route's query parameters. Instead, the setting is set within the
+browser's session storage and the filter applied at query time (rather than
+being passed through via the query params of the page). **Note that this differs
+from the implementation of the "mature" filter that previously existed**. This
+is discussed in further depth in the
+[settings persistence](#settings-persistence) and
+[outstanding questions](#outstanding-questions) sections below.
+
+Sensitive results on the search results page are blurred. Users can unblur and
+reblur individual results through a simple and clear interaction on the search
+results page. In addition to being able to opt-in to having sensitive results
+included in their query, users can disable blurring by default. This parameter
+is stored in the application state. If users disable blurring by default, there
+is not an option to "reblur" images.
+
+Results unblurred by the user on the search results page are also unblurred if
+they visit the single result page for that result. Results that have not been
+unblurred remain blurred upon navigating to the single results page.
+
+If a user lands directly on a single result page for a result with sensitive
+textual content, the result is always be blurred by default. This is true
+regardless of whether the user has disabled blurring by default on the search
+results page, but only applies to page navigation _directly_ to a single results
+page rather than a client side navigation.
+
+#### Settings persistence
+
+Right now, the recommendation is to store the "include sensitive results"
+setting in session storage and the "do not blur sensitive results" setting in
+the ephemeral application state. "Include sensitive settings" should not be
+reset by a page reload, but "do not blur sensitive results" should. To
+summarise:
+
+| Setting                       | Default | Persistence location            | Behaviour implications                                                                                                                                              |
+| ----------------------------- | ------- | ------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Include sensitive results     | Off     | Session storage                 | Cannot be manipulated through the query params. Does not reset on page reload or in new tabs. Does reset when the browser is restarted or when the session expires. |
```

Comment:
Thank you for explaining the reasons behind not wanting to use the query param, it makes total sense.

By "Session storage", do you mean a session that is stored on the server, or `Window.sessionStorage`?

Reply:
I think you're asking about the technical implementation of a session-stored variable for this (apologies if I am misunderstanding). My understanding is that it will need to be a cookie for the technical implementation to work as the server will need to know on SSR requests whether to send the `include_sensitive_results` parameter.

Here is the MDN documentation about this particular cookie lifetime setting: https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#define_the_lifetime_of_a_cookie

Reply:
Thank you for clarification! I was thrown off by the name resemblance to `Window.sessionStorage`, which is reset between tabs and is more ephemeral than `localStorage`. I see that you weren't describing that.

Reply:
I should change it to "Session cookie" to avoid the confusion, thank you!